### PR TITLE
feat: Set default to null when the field type is nullable (activable by the configuration)

### PR DESCRIPTION
**Pull request by Chuckame**
Quick PR that have no impact on the existing features. I hope you will release it before the work of primitives in root :pray:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -415,10 +415,33 @@ Would result in the following schema:
 }
 ```
 
+### Nullable fields, optional fields and compatibility
```
Comment (thake) [inline README.md:418]: Great! Thanks for the good documentation. This is often something that is overlooked when opening up PRs.

File: src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt [hunk 2]
```diff
@@ -115,18 +116,20 @@ class ClassSchemaFor(
          else -> schemaOrFixed.overrideNamespace(ns)
       }
 
-      val default: Any? = annos.default()?.let {
+      val default: Any? = annos.default()?.let { annotationDefaultValue ->
```
Comment (thake) [inline src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt:119]: Maybe we should extract the logic for determining the default value to a separate function. With the added if clause, it becomes hard to read.

**Review by thake — CHANGES_REQUESTED** [review]
Thanks for the PR a really useful addition. It will certainly be merged before primitive types.

**Comment by thake** [comment]
@Chuckame, do you want to join me in maintaining this repo?

File: src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt (hunk 2 above)
Comment (Chuckame) [reply src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt:119]: Good point, I will do it!

**Comment by Chuckame** [comment]
@thake yes, why not!!

**Comment by Chuckame** [comment]
> Thanks for the PR a really useful addition. It will certainly be merged before primitive types.

Good idea, did you worked on it since the last commit to your primitives in root branch?

**Comment by thake** [comment]
> > Thanks for the PR a really useful addition. It will certainly be merged before primitive types.
> 
> Good idea, did you worked on it since the last commit to your primitives in root branch?

No, not yet. If you want, you can give it a try.

**Comment by Chuckame** [comment]
Yes I begun something similar to Json, proto, and other native serializers. I would like to lighten the Avro class, make some extensions and make it more configurable (starting with this pr)

**Comment by Chuckame** [comment]
By the way, how do you want me to make releases? That way we add a CONTRIBUTING.md 😁

**Comment by thake** [comment]
> By the way, how do you want me to make releases? That way we add a CONTRIBUTING.md grin

I've opened up an issue #143 for that. Maybe we'll have to change something on the release workflow.

**Comment by thake** [comment]
> Yes I begun something similar to Json, proto, and other native serializers. I would like to lighten the Avro class, make some extensions and make it more configurable (starting with this pr)

That sounds really awesome :) Do you want to talk me through before? That way I can give you some input and work out a solution together. You can find my contact details on my github profile, I think we should discuss this "face to face".

File: src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt (hunk 2 above)
Comment (Chuckame) [reply src/main/kotlin/com/github/avrokotlin/avro4k/schema/ClassSchemaFor.kt]: Done

**Pull request merged**